agent/eventhandler/task_handler.go

```diff
@@ -260,7 +279,24 @@ func (handler *TaskHandler) flushBatchUnsafe(taskStateChange *api.TaskStateChang
 	// All container events for the task have now been copied to the
 	// task state change object. Remove them from the map
 	delete(handler.tasksToContainerStates, taskStateChange.TaskARN)
-
+	taskStateChange.ManagedAgents = append(taskStateChange.ManagedAgents,
+		handler.tasksToManagedAgentStates[taskStateChange.TaskARN]...)
+	// All managed agent events for the task have now been copied to the
+	// task state change object. Remove them from the map
+	delete(handler.tasksToManagedAgentStates, taskStateChange.TaskARN)
+	if taskStateChange.Task.GetKnownStatus() >= apitaskstatus.TaskStopped {
```

Comment:
We need to send status stopped for managed agent when the container is stopped, regardless of the task state. This is not going to send stopped when task is running but containers are stopped (which I believe can only happen with non essential containers)

Reply:
missed that, will change

Reply:
removed this change. have added a TODO and will address when model changes are in.